### ISSUE TITLE
1B-4: CSS validation TS integration via napi-rs

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3490]
Parent: #3178 (Phase 1B-4)
Depends on: #3475 (1B-1, ✅), #3478 (1B-2, ✅), #3488 (1B-3, ✅)

## Overview

Expose the Rust CSS value matching engine via napi-rs so that it is callable from Node.js. This completes Phase 1B — all CSS validation logic is available in Rust and accessible from the JS runtime.

**This phase does NOT replace the TS `@markuplint/types` package.** The actual switchover (rules calling `@markuplint/core` instead of `@markuplint/types`) is Phase 8 work.

### 1. napi-rs functions

Add to `crates/markuplint-napi/src/lib.rs`:

- `matchCssSyntax(syntax: string, value: string) → CssMatchResult`
- `matchCssProperty(syntax: string, value: string) → CssMatchResult`

```typescript
interface CssMatchResult {
  matched: boolean;
  offset?: number;    // mismatch byte offset
  length?: number;    // mismatch length
  expected?: string[]; // expected values at mismatch point
}
```

### 2. CI verification

- napi binary builds on all 3 platforms (ubuntu, macos, windows) — verified by existing `napi-smoke` CI job
- Rust tests pass (`cargo test -p markuplint-types`, `cargo clippy -- -D warnings`)

### 3. Smoke test

- JS-level test that loads the napi binary and calls `matchCssSyntax` / `matchCssProperty`
- Verifies basic matching: keyword, type, CSS-wide keywords, var(), calc()

## Out of scope (deferred to Phase 8)

The following tasks are **explicitly deferred** and must be tracked in Phase 8:

### 8-A. Replace `@markuplint/types` CSS validation path

- `@markuplint/rules` currently calls `@markuplint/types` → `cssSyntaxMatch()` → `css-tree`
- In Phase 8, change to: `@markuplint/rules` → `@markuplint/core` (napi) → `markuplint-types` (Rust)
- `@markuplint/types` (TS package) becomes obsolete for CSS validation
- `css-tree` dependency can be removed

### 8-B. Non-CSS type validation migration

- `@markuplint/types` also handles non-CSS types (DateTime, AutoComplete, LinkType, etc.)
- These are already implemented in Rust (`markuplint-types` crate, Phase 1A-3)
- Phase 8 should migrate all type validation calls to `@markuplint/core`

### 8-C. `@markuplint/types` package disposition

- Once all consumers switch to `@markuplint/core`, `@markuplint/types` can be:
  - Deprecated (re-export from `@markuplint/core` for backward compat)
  - Or removed entirely (v6 breaking change is acceptable)
- TS type definitions (`Type`, `Result`, `Defs`, etc.) may need to stay as public API types

### 8-D. `@markuplint/ml-spec` integration

- `@markuplint/ml-spec` currently imports types from `@markuplint/types`
- Phase 8 must update this to use `@markuplint/core` or keep type-only imports

## Architecture: Phase 1B-4 vs Phase 8

```
Phase 1B-4 (this issue): Make Rust callable from JS
──────────────────────────────────────────────────
  @markuplint/core (napi) ← matchCssSyntax(), matchCssProperty()
       ↓
  markuplint-types (Rust) ← value_match engine + custom types

Phase 8: Switch callers to Rust
──────────────────────────────
  @markuplint/rules ──→ @markuplint/core (napi) ──→ markuplint-types (Rust)
       ↓ (removed)
  @markuplint/types (TS) ──→ css-tree  ← REMOVED
```

## Reference

- `crates/markuplint-napi/src/lib.rs` — napi entry point
- `crates/markuplint-types/src/css/value_match/mod.rs` — Rust public API
- `packages/@markuplint/core/package.json` — napi binary wrapper
- `packages/@markuplint/types/src/css-syntax.ts` — Current css-tree integration (Phase 8 replacement target)

**Comment by YusukeHirao** [comment]
Completed in PR #3493. CSS value matching engine exposed via napi-rs: matchCssSyntax() and matchCssProperty(). Phase 1B complete.
